File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    a, b = 0, 0
    while s != '1' :
        t = len(s)
        s = s.replace("0", "")
        b += t - len(s)
        
        s = bin(len(s))[2:]
        
        a += 1
    
    return [a, b]

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        int[] answer = {0, 0};
        
        while (!s.equals("1")) {
            answer[0] += 1;
            int first = s.length();
            s = s.replaceAll("0", "");
            int second = s.length();
            answer[1] += first - second;
            s = bin(s.length());
        }
        return answer;
    }
    
    public String bin(int s) {
        return Integer.toBinaryString(s);
    }
}